be/src/olap/tablet.cpp

```diff
@@ -713,21 +713,6 @@ bool Tablet::can_do_compaction(size_t path_hash, CompactionType compaction_type)
         return false;
     }
 
-    if (tablet_state() == TABLET_RUNNING) {
-        // if tablet state is running, we need to check if it has consistent versions.
-        // tablet in other state such as TABLET_NOTREADY may not have complete versions.
-        std::shared_lock rdlock(_meta_lock);
-        const RowsetSharedPtr lastest_delta = rowset_with_max_version();
```

Comment:
the `rowset_with_max_version` method checked un-published version, we'd better change the implementation here.
e.g. if the published rowset have missing version, we should cancel compaction.

Reply:
in concurrent load, this check is not correct cause some publish task is flying, remove this check because we have consective version check in pick input rowset